sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/GroupIntoBatches.java

```diff
@@ -48,25 +48,24 @@
  * are output to the output {@link PCollection}.
  *
  * <p>Windows are preserved (batches contain elements from the same window). Batches may contain
- * elements from more than one bundle
+ * elements from more than one bundle.
  *
- * <p>Example (batch call a webservice and get return codes)
+ * <p>Example (batch call a webservice and get return codes):
  *
  * <pre>{@code
- *  Pipeline pipeline = Pipeline.create(...);
- *  ... // KV collection
- *  long batchSize = 100L;
- *  pipeline.apply(GroupIntoBatches.<String, String>ofSize(batchSize))
- * .setCoder(KvCoder.of(StringUtf8Coder.of(), IterableCoder.of(StringUtf8Coder.of())))
- * .apply(ParDo.of(new DoFn<KV<String, Iterable<String>>, KV<String, String>>() {
- * {@literal @}ProcessElement
- * public void processElement({@literal @}Element KV<String, Iterable<String>> element,
- *                            OutputReceiver<KV<String, String>> r) {
- * r.output(KV.of(element.getKey(), callWebService(element.getValue())));
- * }
- * }));
- *  pipeline.run();
- * }</pre>
+ * PCollection<KV<String, String>> input = ...;
+ * long batchSize = 100L;
+ * PCollection<KV<String, Iterable<String>>> batched = input
+ *     .apply(GroupIntoBatches.<String, String>ofSize(batchSize))
+ *     .setCoder(KvCoder.of(StringUtf8Coder.of(), IterableCoder.of(StringUtf8Coder.of())))
+ *     .apply(ParDo.of(new DoFn<KV<String, Iterable<String>>, KV<String, String>>() }{
+ *        {@code @ProcessElement
```

Comment:
On this line, `@` would be the first non-whitespace character if we weren't starting a new `@code` block, which would cause Javadoc to raise an error.

The solution here is to end the `@code` block and start a new one on this line. Note that `@code` and `@literal` blocks, while they will render literal curly brackets, expect curly brackets to be balanced. Thus, we have to end the previous code block before the `{` that opens the `DoFn` body in our example.

Thus, line 61 here contains `}{` to end the code block and render a literal unmatched `{`. Then, we start a new code block on this line and we close this block on line 67 so that we can have a literal unmatched `}` closing the `DoFn` body of the example.

Reply:
I've filed https://issues.apache.org/jira/browse/BEAM-6679 to look at fixing up `@literal` in other classes.

Reply:
Note that an alternative to the method I'm using here would be to wrap the entire example only in `<pre>`, and then apply `{@code}` only to smaller sections of the example where angle brackets or `@` show up. The complication there is that `{@code x}` always inserts a space before the text `x`, so there's fiddling necessary to decide where the tag should start so that there's a natural space, and to remove a preceeding space so alignment stays correct.